engines/private/private.cpp

```diff
@@ -120,40 +121,67 @@ void PrivateEngine::initializePath(const Common::FSNode &gamePath) {
 	SearchMan.addDirectory(gamePath.getPath(), gamePath, 0, 10);
 }
 
-Common::Error PrivateEngine::run() {
+Common::SeekableReadStream *PrivateEngine::loadAssets() {
 
-	assert(_installerArchive.open("SUPPORT/ASSETS.Z"));
+	Common::File *test = new Common::File();
 	Common::SeekableReadStream *file = NULL;
-	// if the full game is used
-	if (!isDemo()) {
-		assert(_installerArchive.hasFile("GAME.DAT"));
-		file = _installerArchive.createReadStreamForMember("GAME.DAT");
+
+	if (isDemo() && test->open("SUPPORT/ASSETS/DEMOGAME.WIN"))
+		file = (Common::SeekableReadStream *) test;
+	else if (test->open("SUPPORT/ASSETS/GAME.WIN")) {
+		file = (Common::SeekableReadStream *) test;
 	} else {
-		// if the demo from archive.org is used
-		if (_installerArchive.hasFile("GAME.TXT"))
-			file = _installerArchive.createReadStreamForMember("GAME.TXT");
-
-		// if the demo from the full retail CDROM is used
-		else {
-			if (_installerArchive.hasFile("DEMOGAME.DAT"))
-				file = _installerArchive.createReadStreamForMember("DEMOGAME.DAT");
+		delete test;
+		assert(_installerArchive.open("SUPPORT/ASSETS.Z"));
+		// if the full game is used
+		if (!isDemo()) {
+			if (_installerArchive.hasFile("GAME.DAT"))
+				file = _installerArchive.createReadStreamForMember("GAME.DAT");
+			else if (_installerArchive.hasFile("GAME.WIN")) 
+				file = _installerArchive.createReadStreamForMember("GAME.WIN");
+			else
+				error("Unknown version");
+		} else {
+			// if the demo from archive.org is used
+			if (_installerArchive.hasFile("GAME.TXT"))
+				file = _installerArchive.createReadStreamForMember("GAME.TXT");
+
+			// if the demo from the full retail CDROM is used
+			else if (_installerArchive.hasFile("DEMOGAME.DAT"))
+					file = _installerArchive.createReadStreamForMember("DEMOGAME.DAT");
+			else if (_installerArchive.hasFile("DEMOGAME.WIN"))
+					file = _installerArchive.createReadStreamForMember("DEMOGAME.WIN");
+			else {
+				error("Unknown version");
+			}
 		}
-	}
+	}	
+	assert(file != NULL);
+	return file;
 
+}
+
+Common::Error PrivateEngine::run() {
+
+	Common::SeekableReadStream *file = loadAssets();
 	// Read assets file
-	assert(file != NULL);
-	const int32 fileSize = file->size();
+	const uint32 fileSize = file->size();
 	char *buf = (char *)malloc(fileSize + 1);
 	file->read(buf, fileSize);
 	buf[fileSize] = '\0';
 
+	Decompiler decomp(buf, fileSize, false);
```

Comment:
I suggest creating another debug channel and then wrap this around like this:

```C++
if (debugChannelSet(-1, kDebugDump)) {
   Decompiler...
```

Reply:
Decompilation is not a debug feature, but a requirement to run localized versions without re-implementing the engine. I still added a debug statement to show the decompiled code, in the `code` debug channel.